astropy/config/paths.py

```diff
@@ -165,14 +166,33 @@ def wrapper(*args: P.args, **kwargs: P.kwargs) -> None:
 
     @classmethod
     def _get_dir_path(cls, rootname: str) -> Path:
+        if (usrpth_str := os.getenv(cls._astropy_env_dir)) is not None:
+            usrpth = Path(usrpth_str).expanduser()
+
+            if usrpth.is_file():
+                raise FileExistsError(
+                    f"Cannot create directory under {usrpth_str} as requested "
+                    f"via the environment variable {cls._astropy_env_dir}: "
+                    "a file with this name already exists."
+                )
+
+            if not usrpth.parent.exists():
+                raise FileNotFoundError(
+                    f"Cannot create directory under {usrpth_str} as requested "
+                    f"via the environment variable {cls._astropy_env_dir}: "
+                    f"the parent directory {usrpth.parent} does not exist."
+                )
+            usrpth.mkdir(exist_ok=True)
+            return usrpth.resolve()
```

Comment:
It would be simplest both for us and for the users if the effect of the new Astropy-specific environment variables would be as similar as possible to the effect of the already available XDG environment variables. My (naive) expectation is that the code that handles that should look something like
```python
for env_dir in (cls._astropy_env_dir, cls._xdg_env_dir):
  ...
```
Why is there a separate code block for the new functionality?

Reply:
I can't remember what my reasoning was, maybe I was just trying to get the most simple behavior right from the get go; the existing logic for XDG variables is quite complex and I probably thought I couldn't test it all efficiently.